src/Dapr.Actors/Runtime/ActorStateManager.cs

```diff
@@ -392,11 +529,20 @@ private void EnsureStateProviderInitialized()
 
         private sealed class StateMetadata
         {
-            private StateMetadata(object value, Type type, StateChangeKind changeKind)
+            private StateMetadata(object value, Type type, StateChangeKind changeKind, DateTimeOffset? ttlExpireTime = null, TimeSpan? ttl = null)
             {
                 this.Value = value;
                 this.Type = type;
                 this.ChangeKind = changeKind;
+
+                if (ttlExpireTime.HasValue && ttl.HasValue) {
+                    throw new ArgumentException("Cannot specify both TTLExpireTime and TTL");
+                }
+                if (ttl.HasValue) {
+                    this.TTLExpireTime = DateTime.UtcNow.Add(ttl.Value);
```

Comment:
`DateTimeOffset.UtcNow`?  (Doesn't really *need* to be specified in UTC, as sometimes keeping the local time is useful for logging).